enterprise/internal/batches/sources/gitlab.go

```diff
@@ -412,6 +422,36 @@ func readPipelines(it func() ([]*gitlab.Pipeline, error)) ([]*gitlab.Pipeline, e
 	}
 }
 
+func (s *GitLabSource) determineVersion(ctx context.Context) (*semver.Version, error) {
+	var v string
+	chvs, err := versions.GetVersions()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, chv := range chvs {
+		if chv.ExternalServiceKind != extsvc.KindGitLab && chv.Key != s.client.Urn() {
```

Comment:
shouldn't this be the following?
Otherwise this _could_ match on a `GitHub` external service I think. 

```suggestion
		if chv.ExternalServiceKind != extsvc.KindGitLab || chv.Key != s.client.Urn() {
```

Reply:
No, it shouldn't. And yes, it should match GitHub.

The `if ...` block simply contains a `continue` statement because we want to match external services that are of `KindGitlab` and have this same key as the `Urn` of the current client on the Gitlab source.

Reply:
But if type == GitHub && this.URN == Key(), then it could still use that version if I am not completely off. The uniqueness of URN is only guaranteed to be for a given Kind, not globally

Reply:
Can you add a unit test for this scenario?

Reply:
Added more external services to the test suite (to match and not match the `.Key`) and refactored the condition so it's clearer.

Thanks